tests/Microsoft.Identity.Test.Integration.netcore/HeadlessTests/ClientCredentialsMtlsPopTests.cs

```diff
@@ -0,0 +1,109 @@
+        {
+            TestCommon.ResetInternalStaticCaches();
+        }
+
+        [TestMethod]
+        [DataRow("https://login.microsoftonline.com/bea21ebe-8b64-4d06-9f6d-6a889b120a7c", DisplayName = "Standard Authority")]
+        [DataRow("https://mtlsauth.microsoft.com/bea21ebe-8b64-4d06-9f6d-6a889b120a7c", DisplayName = "MTLS Authority")]
+        public async Task Sni_Gets_Pop_Token_Successfully_TestAsync(string authority)
+        {
+            // Arrange: Use the public cloud settings for testing
+            IConfidentialAppSettings settings = ConfidentialAppSettings.GetSettings(Cloud.Public);
+
+            // Retrieve the certificate from settings
+            X509Certificate2 cert = settings.GetCertificate();
+
+            // Build Confidential Client Application with SNI certificate at App level
+            IConfidentialClientApplication confidentialApp = ConfidentialClientApplicationBuilder.Create(MsiAllowListedAppIdforSNI)
+                .WithAuthority(authority)
+                .WithAzureRegion("westus3") //test slice region 
+                .WithCertificate(cert, true)  // Configure SNI certificate at App level
+                .WithExperimentalFeatures()
+                .WithTestLogging()
+                .Build();
+
+            // Act: Acquire token with MTLS Proof of Possession at Request level
+            AuthenticationResult authResult = await confidentialApp
+                .AcquireTokenForClient(settings.AppScopes)
+                .WithMtlsProofOfPossession()
+                .WithExtraQueryParameters("dc=ESTSR-PUB-WUS3-AZ1-TEST1&slice=TestSlice") //Feature in test slice 
+                .WithSendX5C(true)
+                .ExecuteAsync()
+                .ConfigureAwait(false);
+
+            // Assert: Check that the MTLS PoP token acquisition was successful
+            Assert.IsNotNull(authResult, "The authentication result should not be null.");
+            Assert.AreEqual(Constants.MtlsPoPTokenType, authResult.TokenType, "Token type should be MTLS PoP");
+            Assert.IsNotNull(authResult.AccessToken, "Access token should not be null");
+
+            // Simulate cache retrieval to verify MTLS configuration is cached properly
+            authResult = await confidentialApp
+               .AcquireTokenForClient(settings.AppScopes)
+               .WithMtlsProofOfPossession()
+               .ExecuteAsync()
+               .ConfigureAwait(false);
+
+            // Assert: Verify that the token was fetched from cache on the second request
+            Assert.AreEqual(TokenSource.Cache, authResult.AuthenticationResultMetadata.TokenSource, "Token should be retrieved from cache");
+        }
+
+        [TestMethod]
+        public async Task SNI_MtlsPopFlow_FailureDueToInvalidConfiguration_TestAsync()
+        {
+            // Arrange: Use the public cloud settings for testing
+            IConfidentialAppSettings settings = ConfidentialAppSettings.GetSettings(Cloud.Public);
```

Comment:
What is the "invalid config" ?

Reply:
added the error and updated the test name, basically a non SNI cert used in this test